src/configuration.cpp

```diff
@@ -728,13 +791,49 @@ bool chimera::CompiledConfiguration::Render(
                   << std::endl;
         return false;
     }
-    std::string mangled_name = ::mstch::render("{{mangled_name}}", context);
+    const std::string mangled_name
+        = ::mstch::render("{{mangled_name}}", context);
+
+    // Create collections for the ordered sets of sources.
+    ::mstch::array binding_sources(parent_.inputSourcePaths_.begin(),
+                                   parent_.inputSourcePaths_.end());
+
+    // Create a top-level context that contains the extracted information
+    // about this particular binding component.
+    ::mstch::map full_context{{key, context}, {"sources", binding_sources}};
+
+    // Resolve customizable snippets that will be inserted into the file
+    // from the configuration file's "template::file" entry.
+    if (bindingNode_)
+    {
+        chimera::util::extendWithYAMLNode(
+            full_context, bindingNode_["file"], false,
+            std::bind(&chimera::CompiledConfiguration::Lookup, this,
+                      std::placeholders::_1));
+    }
+
+    if (not Render(mangled_name, header_view, "h", context, full_context))
+        return false;
+
+    if (not Render(mangled_name, source_view, "cpp", context, full_context))
+        return false;
+
+    // Record this binding name for use at the top-level.
+    binding_names_.push_back(mangled_name);
+    return true;
+}
 
+bool chimera::CompiledConfiguration::Render(
+    const std::string &mangled_name, const std::string &view,
+    const std::string &extension,
+    const std::shared_ptr<::mstch::object> &context,
+    const ::mstch::map &full_context)
+{
```

Comment:
At the beginning of this function, we could check if `view` matches a `NO_RENDER` sentinel value and return immediately.

Reply:
Done https://github.com/personalrobotics/chimera/pull/289/commits/3fb77d63aa41ac4da2e2680fb509ed101dc6a03a